### PR TITLE
Add cache for fetching azure ad users

### PR DESCRIPTION
**Pull request by gavinfish**
Deprecated lazy loading #59  for confused user experience.

Introduced cache to reduce the frequency of fetching data from Azure.

### CODE REVIEW

**Comment by gavinfish** [comment]
@Yatheesh-Rai I have drafted a possible solution for cache here.

**Comment by Yatheesh-Rai** [comment]
Thanks a lot  @gavinfish for the work.

**Comment by gavinfish** [comment]
@Yatheesh-Rai you can try the rc version at [here](https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/azure-ad/1.1.1-rc95.a4bbb3150a90/).

File: src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java [hunk 1]
```diff
@@ -165,12 +184,15 @@ private String getRootUrl() {
...
-    public AzureSecurityRealm(String tenant, String clientId, String clientSecret)
-            throws ExecutionException, IOException, InterruptedException {
+    public AzureSecurityRealm(String tenant, String clientId, String clientSecret, int cacheDuration) {
         super();
         this.clientId = Secret.fromString(clientId);
         this.clientSecret = Secret.fromString(clientSecret);
         this.tenant = Secret.fromString(tenant);
+        this.cacheDuration = cacheDuration;
+        caches = CacheBuilder.newBuilder()
+                .expireAfterWrite(cacheDuration, TimeUnit.HOURS)
```
Comment (lhotari) [inline src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java]: why force cacheDuration to full hours? wouldn't seconds be better for a cache?

**Pull request merged**